s3fs/tests/test_s3fs.py

```diff
@@ -575,6 +575,16 @@ def test_mkdir(s3):
     assert bucket in s3.ls("/")
 
 
+def test_mkdir_existing_bucket(s3):
+    # mkdir called on existing bucket should be no-op and not calling create_bucket
+    # creating a s3 bucket
+    bucket = "test1_bucket"
+    s3.mkdir(bucket)
+    # a second call.
+    s3.mkdir(bucket)
```

Comment:
I would have thought this should raise FileExistsError, no?
The other related methods to check:
- s3.makedirs(bucket, exist_ok=True) # no error
- s3.makedirs(bucket, exist_ok=False) # FileExists error
- s3.makedirs(bucket + "/any/sub/path") # no error
- s3.mkdir(bucket + "/any/other/path", create_parents=True)  # no error

Reply:
you are right... not sure what i was thinking, let me submit another pr, another question though,
if we have 
`
s3.makedirs(existing_bucket + "/path/to/existing/object", exist_ok=False)
`
The current code is no going to raise any error, is this behavior by design too?

Reply:
done, can you review again?